roles/sushy_emulator/tasks/setup_access.yml

```diff
@@ -0,0 +1,55 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Look for Sushy Emulator keypair file
+  register: _sushy_emulator_key_file
+  ansible.builtin.stat:
+    path: "{{ (cifmw_sushy_emulator_resource_directory, 'sushy_emulator_key') | path_join }}"
+
+- name: "Generate SSH keypair"
+  when:
+    - not _sushy_emulator_key_file.stat.exists
+  community.crypto.openssh_keypair:
+    comment: "sushy emulator SSH keypair"
+    path: "{{ (cifmw_sushy_emulator_resource_directory, 'sushy_emulator_key') | path_join }}"
+    type: "ecdsa"
+
+- name: Set authorized key taken from file
+  delegate_to: "{{ cifmw_edpm_deploy_baremetal_hypervisor_target }}"
+  become: true
+  ansible.posix.authorized_key:
+    user: "{{ cifmw_sushy_emulator_libvirt_user }}"
+    state: present
+    key: "{{ lookup('file', cifmw_sushy_emulator_resource_directory+'/sushy_emulator_key.pub') }}"
```

Comment:
the public key is in the returned parameter from the openssh_keypair - you just need to add a `register: _sshkey` to the above task, and then you can access `_sshkey.public_key`:
https://docs.ansible.com/ansible/latest/collections/community/crypto/openssh_keypair_module.html#return-values

Also, beware, `lookup()` is local - so it would probably crash if we run the role from a laptop against a remote hypervisor.

Reply:
The reason I read from the file rather than using the return value is in case the key is already created, the "Generate SSH keypair" task doesn't run, reading from the file works for both cases.

The key files are created locally(AFAIU on the controller) and added to the remote hypervisor so using lookup works in my testing but I'm probably missing a use case.

Reply:
hmm if the key exists, I'd expect it to be in the authorized_keys already?

As I read, the key is created on the remote target - there's no delegate_to on that task. and `lookup()` is running exclusively on the ansible controller (aka your laptop) - it won't be an issue if you're running the role against `localhost` - but... it will crash if it's running against a remote host.

Reply:
> hmm if the key exists, I'd expect it to be in the authorized_keys already?

True! Let me fix that.